<a href="https://colab.research.google.com/github/CeciliaCTorales/misNotebooks/blob/main/Medidas_de_dispersi%C3%B3n.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Objetivos:
- Calcular el rango, la varianza y la desviación standard de las ventas mensuales.
- Reconocer cómo estas medidas de dispersión informan sobre la estbilidad de las ventas.

In [3]:
import numpy as np
import pandas as pd

# generamos un dataset sencillo
ventas = {
    'Mes': ['Enero', 'Febrero', 'Marzo', 'Abril', 'Mayo', 'Junio', 'Julio', 'Agosto', 'Septiembre', 'Octubre', 'Noviembre', 'Diciembre'],
    'ventas (millones)': [1.2, 2.5, 3.1, 18.3, 40.5, 52.1, 54.8, 46.2, 25.5, 13.8, 11.9, 9.2] # Datos de ejemplo en millones
}

df_ventas = pd.DataFrame(ventas)

1. Calcular el rango de las ventas

El rango es la diferencia entre el valor máximo y el mínimo de un conjunto de datos

In [4]:
rango = np.max(df_ventas['ventas (millones)']) - np.min(df_ventas['ventas (millones)'])
print(f'El rango es: {rango}')

El rango es: 53.599999999999994


2. Calcular la varianza de las ventas

La varianza (al igual que la desviación) es una medida de la dispersión de los datos. Sus valores están expresados al cuadrado.
La varianza mide cuan dispersos están los valores, respecto a la media.

In [5]:

varianza = np.var(df_ventas['ventas (millones)'])
print(f'La varianza es: {varianza}')

La varianza es: 368.3224305555555


3 - Calcular la desviación Standard de las ventas
Es la raiz cuadrada de la varianza, proporcionando una medida de dispersión en las mismas unidades que los datos origianles.

In [7]:
desviacion_std = np.std(df_ventas['ventas (millones)'])
print(f'La Desviación Estándard es:  {desviacion_std}')

La Desviación Estándard es:  19.19172818053537


In [13]:
media = np.mean(df_ventas['ventas (millones)'])
CV = desviacion_std / media *100
print(f'El Coeficiente de variación es: {round(CV,2)}')
print(f'La media es: {round(media,2)}')

El Coeficiente de variación es: 82.52
La media es: 23.26


# Análisis
El coeficiente de variación es útil para analizar la desviación Estándar. Es el porcentaje que representa la desviación estándar respecto de la media.
- Una desviación Estándar de 19.19 (cada dato se aleja en promedio 19 millones de la media)
- El Coeficiente de variación del 82% es alto e indica que los datos están muy dispersos respecto de la media.

- Esta información indica que las ventas no son uniformes a lo largo del año, es decir algunos meses tuvieron ventas muy bajas y otros muy altas.